adr/0002-reactive-rename.adoc

```diff
@@ -0,0 +1,47 @@
+= Rename Reactive extensions
+
+* Status: proposed
+* Date: 2023-10-17
+
+== Context and Problem Statement
+
+RESTEasy Reactive is the preferred REST stack in Quarkus, however in the almost 3 years since its release, we have seen
+folks of all experience misunderstand its capabilities and being led to believe that it forces reactive programming on them.
+With the advent of Virtual Threads in Java 19+, the problem becomes larger since in the minds of users, Reactive and Virtual Threads
+are two different worlds.
+
+The same problem applies to other extensions as well:
+* Reactive Messaging
+* Reactive Routes
+* REST Client reactive
+
+The common theme here is that these extensions can work equally well with reactive and blocking code, but their name confuses users.
+
+We have tried to clarify the situation with blog posts, talks and on the various communication channels, but there seems to be
+no escaping the fact that "reactive" in the name primes people in a non-desirable way.
+
+
+== Considered options
+
+We can continue with the current situation, but that feels less than ideal given the recurrence of the problem.
+
+== Decision
+
+* Rename all `quarkus-resteasy-reactive*` maven modules to `quarkus-rest*`
```

Comment:
how does this impact quarkus-rest-client users ?

Reply:
See other comments specifically on the rest-client.

Reply:
the comments don't explain it. ADR should list that quarkus-rest users will be broken and what is the suggested migration path if that is the proposal.

Reply:
Sure, but I wanted us to reach an agreement on what to do with the REST Client before updating the ADR

Reply:
i get that - i'm just saing uou don't need to wait to reach an agreement to highlight that if we rename to quarkus-rest-client then there will be breakage.

Reply:
I've already captured that in the latest version